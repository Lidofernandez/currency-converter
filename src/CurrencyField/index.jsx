import React from "react";
import PropTypes from "prop-types";

import Dropdown from "../Dropdown";

import css from "./index.scss";

const CurrencyField = ({
  currencies,
  selectedCurrency,
  getCurrency,
  value,
  handleOnChange,
  currencyTestid,
}) => (
  <div className={css["currency-field"]}>
    <input
      onChange={handleOnChange}
      value={value}
      type="number"
      className={css["currency-field--input"]}
    />
    <Dropdown
      options={currencies}
      getValue={getCurrency}
      value={selectedCurrency}
      testid={currencyTestid}
    />
  </div>
);

CurrencyField.defaultProps = {
  value: "",
  currencies: [],
  selectedCurrency: "",
  handleOnChange: () => {},
  getCurrency: () => {},
  currencyTestid: "Dropdown",
};

CurrencyField.propTypes = {
  value: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  currencies: PropTypes.arrayOf(PropTypes.string),
  selectedCurrency: PropTypes.string,
  handleOnChange: PropTypes.func,
  getCurrency: PropTypes.func,
  currencyTestid: PropTypes.string,
};

export default CurrencyField;
