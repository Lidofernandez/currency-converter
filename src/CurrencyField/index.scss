.currency-field {
  width: 100%;
  display: flex;
  flex-direction: row;
  min-height: 2rem;
}

.currency-field--input {
  width: 100%;
  box-sizing: border-box;
}
