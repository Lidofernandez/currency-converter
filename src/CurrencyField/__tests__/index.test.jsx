import React from "react";
import { render, fireEvent } from "@testing-library/react";
import CurrencyField from "../index";

describe("CurrencyField", () => {
  it("should render", () => {
    const { asFragment, getByDisplayValue } = render(
      <CurrencyField value="1" currencies={["EUR", "USD"]} />,
    );

    fireEvent.change(getByDisplayValue("1"), {
      target: { value: "mocked-value" },
    });

    fireEvent.change(getByDisplayValue("EUR"), {
      target: { value: "mocked-value" },
    });

    expect(asFragment()).toMatchSnapshot();
  });

  it("should handle value inputs", () => {
    const handleOnChange = jest.fn();
    const { getByDisplayValue } = render(
      <CurrencyField handleOnChange={handleOnChange} value="1" />,
    );
    fireEvent.change(getByDisplayValue("1"), {
      target: { value: "mocked-value" },
    });

    expect(handleOnChange).toHaveBeenCalledTimes(1);
  });

  it("should render currencies", () => {
    const { getByDisplayValue, getByText } = render(
      <CurrencyField currencies={["EUR", "USD"]} />,
    );
    getByDisplayValue("EUR");
    getByText("USD");
  });
});
