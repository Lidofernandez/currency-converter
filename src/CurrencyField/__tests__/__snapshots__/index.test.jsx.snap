// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyField should render 1`] = `
<DocumentFragment>
  <div>
    <input
      type="number"
      value="1"
    />
    <select
      data-testid="Dropdown"
    >
      <option
        value="EUR"
      >
        EUR
      </option>
      <option
        value="USD"
      >
        USD
      </option>
    </select>
  </div>
</DocumentFragment>
`;
